### PR TITLE
add `constraints` for new framework

### PR DESCRIPTION
**Pull request by pwzgorilla**

### CODE REVIEW

File: src/manager/framework/state/offer_wrapper.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ type OfferWrapper struct {
 	MemUsed      float64
 	DiskUsed     float64
 	PortUsedSize int
+	Attributes   []*mesos.Attribute
```
Comment (cmingxu) [inline src/manager/framework/state/offer_wrapper.go]: leave this on Offer object is good, as OfferWrapper suppose to store intermediate calculated value.

File: src/manager/framework/state/slot.go [hunk 2]
```diff
@@ -154,11 +155,71 @@ func (slot *Slot) Update(version *types.Version) {
 }
 
 func (slot *Slot) TestOfferMatch(ow *OfferWrapper) bool {
+	if slot.Version.Constraints != nil && len(slot.Version.Constraints) > 0 {
+		constraints := slot.filterConstraints(slot.Version.Constraints)
+		for _, constraint := range constraints {
+			cons := strings.Split(constraint, ":")
+			if cons[1] == "LIKE" {
+				for _, attr := range ow.Attributes {
```
Comment (cmingxu) [inline src/manager/framework/state/slot.go]: ow.Offer.Attributes ?

**Comment by cmingxu** [comment]
LGTM but still have room for improvement.

File: src/manager/framework/state/offer_wrapper.go (hunk 1 above)
Comment (pwzgorilla) [inline src/manager/framework/state/offer_wrapper.go]: 噢 没注意还有个offer字段，那这样的话其实就不用单独写一个Attributes字段了，直接从offer就能取到了

File: src/manager/framework/state/slot.go (hunk 2 above)
Comment (pwzgorilla) [inline src/manager/framework/state/slot.go]: 嗯，应该是这样，没注意到那个Offer字段。

**Comment by upccup** [comment]
LGTM

**Pull request merged**